app/api_1_0/url.py

```diff
@@ -0,0 +1,399 @@
+                     unavailable)
+from .validators import LongUrlValidator, ShortUrlValidator
+
+dotenv.load()
+
+
+@api.route('/shorten', methods=['POST'])
+def shorten():
+    """
+    This function accepts a json object from the request body which contains
+    a URL. The URL is shortened and saved in the database returning the
+    details of the shortened URL.
+    """
+    long_url_input = LongUrlValidator(request)
+    if not long_url_input.validate():
+        if 'This field is required.' in long_url_input.errors:
+            return bad_request('A long URL is required')
+        return bad_request(long_url_input.errors)
+    long_url = request.json.get('long_url')
+    search_long_url = LongUrl.query.filter_by(long_url=long_url).first()
+    if search_long_url:
+        short_url = search_long_url.short_urls.filter_by(
+            user=g.current_user).first()
+        if short_url and request.json.get('vanity'):
+            return jsonify({'success': True,
+                            'url': short_url.get_details(),
+                            'info': 'Url shortened by you before'})
+        if short_url:
+            return jsonify({'success': True,
+                            'url': short_url.get_details(),
+                            })
+        new_short_url = get_short_url()
+        return update_new_short_url(new_short_url, search_long_url)
+    new_short_url = get_short_url()
+    return add_new_short_url(new_short_url, long_url)
+
+
+def update_new_short_url(new_short_url, long_url):
+    """
+    Return a shortened URL details after linking it to a long URL.
+
+    Keyword arguments:
+    new_short_url -- the newly generated shortened URL
+    long_url -- the existing long URL
+    """
+    long_url.short_urls.append(new_short_url)
+    g.current_user.short_urls.append(new_short_url)
+    new_short_url.save()
+    return jsonify({'success': True,
+                    'url': new_short_url.get_details()
+                    })
+
+
+def add_new_short_url(new_short_url, long_url):
+    """
+    Return a shortened URL details after saving it along its long URL.
+
+    Keyword arguments:
+    new_short_url -- the newly generated shortened URL
+    long_url -- the new long URL to be linked and saved with the shortened URL
+    """
+    new_short_url.save()
+    new_long_url = LongUrl(long_url=long_url)
+    new_long_url.short_urls.append(new_short_url)
+    g.current_user.short_urls.append(new_short_url)
+    new_long_url.save()
+    return jsonify({'success': True,
+                    'url': new_short_url.get_details()
+                    })
+
+
+def generate_short_url(len_range=5):
+    """
+    This is a recursive function that generates a random string of length
+    len_range.
+
+    Keyword arguments:
+    len_range -- the integer value used to determine the length of the string
+                 it is given a default value of 5
+    """
+    url_prefix = dotenv.get('URL_PREFIX')
```

Comment:
every time this function calls itself it does this thing of getting the prefix.. Is there a way this fetching of the prefix can be DRYied up?